### ISSUE TITLE
problem in save when i try to signup

### ISSUE THREAD

**Issue by sararabie** [rezaies/moodle-profilefield_conditional#2]
i use moodle 3.2 when i use this plugin to hide my custom field debend on another field it worked will and hide my fields and show required field but error in saving when i try to sign up it save all data in user table and when it try to save remain field in table user_info_data  
Error:
Debug info: Column 'dataformat' cannot be null
INSERT INTO mdl_user_info_data (dataformat,userid,fieldid,data) VALUES(?,?,?,?)
 and when go to admin and list users i found this user but the condition field is empty and whent edit it as admin it save without error and when i try to add new user from admin it also save will without error 

my custom fields types
interested courses is text editor or textarea  
where you hear -->text editor
and other field is text
and error display in text editor field although was hidden field but  when not make any text editor hidden it work fine without error **the problem when there is hidden field text editor** 

this is steps with image
![error](https://cloud.githubusercontent.com/assets/13334378/24446903/e4fa2d74-1477-11e7-9e87-bfe6ad1f6f7f.png)

![aftersave_admin](https://cloud.githubusercontent.com/assets/13334378/24446733/3e308088-1477-11e7-94a0-7623254457c2.png)
![screenshot from 2017-03-29 10-25-17](https://cloud.githubusercontent.com/assets/13334378/24446760/57bb73d2-1477-11e7-9ea6-21f9e2b0aeac.png)

**Comment by sararabie** [comment]
are you solve this issue ?? because i need to use this plugin in my website because it will save me alot of time rather than jquery and css

**Comment by rezaies** [comment]
Hi Sara,
This is caused by an inconsistency in Moodle code, therefore it's not easy to handle this case in the plugin.
You can fix it by changing the content of getValue method in lib/form/editor.php to the following:
   ```
    function getValue() {
        foreach ($this->_values as $value) {
            if (!is_null($value)) {
                return $this->_values;
            }
        }
        return null;
    }
```

**Comment by sararabie** [comment]
thank you but this mean i must edit in core files

**Comment by rezaies** [comment]
Sadly yes. This issue is the result of a core bug. So you can either wait for it to be hopefully fixed in later versions or take care of the fix yourself.

**Comment by sararabie** [comment]
ok but there is another issue for me when select field required it add rule required and form not submit till add value in it but i dont need it required i need it show only without rule required
how remove required rule from plugin

**Comment by rezaies** [comment]
I'm not sure if I understand you correctly. If you don't wan't a field to be mandatory, you don't need to set it as a required field. A field would be displayed as long as it is not set as hidden.
Please refer to:
EN: [http://kb.foodle.org/content/2/28/en/conditional-profile-field.html#Configuring conditions](http%3A%2F%2Fkb.foodle.org%2Fcontent%2F2%2F28%2Fen%2Fconditional-profile-field.html%23Configuring+conditions)
FA: [http://kb.foodle.org/content/2/28/fa/منوی-بازشوندهٔ-شرطی.html#پیکربندی شرایط](http%3A%2F%2Fkb.foodle.org%2Fcontent%2F2%2F28%2Ffa%2F%D9%85%D9%86%D9%88%DB%8C-%D8%A8%D8%A7%D8%B2%D8%B4%D9%88%D9%86%D8%AF%D9%87%D9%94-%D8%B4%D8%B1%D8%B7%DB%8C.html%23%D9%BE%DB%8C%DA%A9%D8%B1%D8%A8%D9%86%D8%AF%DB%8C+%D8%B4%D8%B1%D8%A7%DB%8C%D8%B7)

**Comment by sararabie** [comment]
yes i understand but my problem is when i create custom field example call Name is required in creating it and another field called Age not required and field called birthdate not required but i need it show in signup page and i use conditional field for types of user to show some field for this type and some field for another type  i need for first type show field Name and Age
and for another field show birthdate and Name And Age
when use required and hidden  i see required add rule required for this field but i nedd it show only without add new rule for it because i create this custom field not required .

and when i try conditional field for 20 field or more it release error and not work well

**Comment by rezaies** [comment]
Can you please post a screenshot of your condition configuration.
I barely understand your English. Could you please describe in Persian.

**Comment by rezaies** [comment]
This issue has been inactive for a significant period of time, and there hasn't been any recent update or response. I'm going to close this issue for now.
